### PR TITLE
Added keccak256 tests for Hasher

### LINKED ISSUE #5090: Add keccak256 tests for `Hasher`
Currently, only the `sha256` tests exist for the `Hasher` type.

### PR DESCRIPTION
**Pull request by brandonsurh**
## Description
Closes #5090 

## Checklist

- [x] I have linked to any relevant issues.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have updated the documentation where relevant (API docs, the reference, and the Sway book).
- [x] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added (or requested a maintainer to add) the necessary `Breaking*` or `New Feature` labels where relevant.
- [x] I have done my best to ensure that my PR adheres to [the Fuel Labs Code Review Standards](https://github.com/FuelLabs/rfcs/blob/master/text/code-standards/external-contributors.md).
- [ ] I have requested a review from the relevant team or maintainers.

### CODE REVIEW

**Comment by brandonsurh** [comment]
The Check Links test failed due to the site docs.rs going down during the check. It needs to be run again. Also, I don't have permissions to add assignees, reviewers, or labels.

**Review by bitzoic — APPROVED** [review]
Looks great, thanks for contributing! :shipit:

File: sway-lib-std/src/hash.sw [hunk 1]
```diff
@@ -500,3 +500,122 @@ fn test_hasher_sha256_bytes() {
     let sha256 = hasher.sha256();
     assert(sha256 == 0x4bf5122f344554c53bde2ebb8cd2b7e3d1600ad631c385a5d7cce23c7785459a);
 }
+
+// The hashes for the following test can be obtained in Rust by running the following script:
+// https://www.rustexplorer.com/b/kbg0oc
+
```
Comment (Braqzen) [inline sway-lib-std/src/hash.sw:506]:
```suggestion
```

This is a thoughtful addition; however, I think we should not be linking external sites in the source code. Particularly, if they do not belong to us and if someone may easily search for a site to replicate the hash of a value.

File: sway-lib-std/src/hash.sw [hunk 2]
```diff
@@ -500,3 +500,122 @@ fn test_hasher_sha256_bytes() {
...
+    let keccak256 = hasher.keccak256();
+    assert(keccak256 == 0xbc36789e7a1e281436464229828f817d6612f7b477d66591ff96a9e064bcc98a);
+
+    let mut hasher = Hasher::new();
+    let mut bytes = Bytes::new();
+    bytes.push(1u8);
+    bytes.hash(hasher);
+    let keccak256 = hasher.keccak256();
+    assert(keccak256 == 0x5fe7f977e71dba2ea1a68e21057beebb9be2ac30c6410aa38d4f3fbe41dcffd2);
+}
```
Comment (Braqzen) [inline sway-lib-std/src/hash.sw:621]:
```suggestion
    let mut bytes = Bytes::with_capacity(1);
    bytes.push(0u8);
    bytes.hash(hasher);
    let keccak256 = hasher.keccak256();
    assert(keccak256 == 0xbc36789e7a1e281436464229828f817d6612f7b477d66591ff96a9e064bcc98a);

    let mut hasher = Hasher::new();
    let mut bytes = Bytes::with_capacity(1);
    bytes.push(1u8);
    bytes.hash(hasher);
    let keccak256 = hasher.keccak256();
    assert(keccak256 == 0x5fe7f977e71dba2ea1a68e21057beebb9be2ac30c6410aa38d4f3fbe41dcffd2);
}

```

**Review by Braqzen — COMMENTED** [review]
I haven't verified that the resulting hashes are correct so I'm trusting the implementation to produce the correct hashes

File: sway-lib-std/src/hash.sw (hunk 1 above)
Comment (brandonsurh) [reply sway-lib-std/src/hash.sw]: Got it, that makes sense. 👍

**Review by bitzoic — APPROVED** [review]

**Review by Braqzen — APPROVED** [review]

**Pull request merged**